src/runtime/analytics-service.js

```diff
@@ -62,18 +65,30 @@ export class AnalyticsService {
      */
     this.context_ = new AnalyticsContext();
 
-    /**
-     * @private @const {!TransactionId}
-     */
-    this.xid_ = new TransactionId(deps);
-
     /** @private {?Promise<!web-activities/activity-ports.ActivityIframePort>} */
     this.serviceReady_ = null;
 
     /** @private {?Promise} */
     this.lastAction_ = null;
   }
 
+  /**
+   * @param {string} transactionId
+   * @param {boolean=} override
+   */
+  setTransactionId(transactionId, override = false) {
```

Comment:
Still not quire sure about this. Why would one ever call `setTransactionId` w/o `override = true`? Alternatively, we could also drop `this.transactionId_` and always just use `this.context_.get/setTransactionId()`